core/sql/generator/GenPreCode.cpp

```diff
@@ -10033,17 +10033,56 @@ ItemExpr * MathFunc::preCodeGen(Generator * generator)
   if (nodeIsPreCodeGenned())
     return this;
 
+  // for ROUND, if the first operand is a BigNum, don't cast
+  // the children to DOUBLE PRECISION; but do make sure the
+  // second operand is an integer
+  NABoolean castIt = TRUE;
+  if (getOperatorType() == ITM_ROUND)
+    {
+      const NAType &typ0 = child(0)->getValueId().getType();
+      if (((const NumericType &)typ0).isBigNum())
+        {
+          castIt = FALSE;
+
+          if (getArity() > 1)
+            {
+              const NumericType &typ1 = (const NumericType &)child(1)->getValueId().getType();
+              if (!typ1.isInteger() || 
+                  ((typ1.getFSDatatype() != REC_BIN8_SIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN8_UNSIGNED) && 
+                   (typ1.getFSDatatype() != REC_BIN16_SIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN16_UNSIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN32_SIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN32_UNSIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN64_SIGNED) &&
+                   (typ1.getFSDatatype() != REC_BIN64_UNSIGNED) ) )
+                {
```

Comment:
Could use isBinaryNumeric method from dfs2rec class  unless these
specific checks are needed

Reply:
Didn't know about this method. Just took a look at it. It includes REC_BPINT_UNSIGNED, which I had not added run-time support for. I'll add it and make this change.

Reply:
Done! Thanks for the suggestion.